fluentd/vendored_gem_src/fluentd/lib/fluent/plugin/in_tail.rb

```diff
@@ -553,12 +565,28 @@ def on_notify
           stat = nil
         end
 
-        @rotate_handler.on_notify(stat) if @rotate_handler
+        @totalbytesavailable=@rotate_handler.on_notify(stat,@inodelastfsize_map) if @rotate_handler
         @line_buffer_timer_flusher.on_notify(self) if @line_buffer_timer_flusher
-        @io_handler.on_notify if @io_handler
+        @io_handler.on_notify(@inodereadfsize_map) if @io_handler
+
+        @logloss=0
+        if @io_handler && @rotate_handler
+          @totalbytesread=0
+          @countonrotate=0
+          @inodereadfsize_map.each do |k, v|
+            @totalbytesread+=v
+            @countonrotate+=1
+          end
+
+        #@log.info "inodelastfsize_map #{@inodelastfsize_map} inodereadfsize_map #{@inodereadfsize_map}"
+        @logloss=@totalbytesavailable-@totalbytesread
+        #@log.info "totalbytesavailable #{@totalbytesavailable} totalbytesread #{@totalbytesread} logloss #{@logloss} countonrotate #{@countonrotate}"
+        end  
+
       end
 
       def on_rotate(stat)
+        @maxfsize=@pe.read_pos
```

Comment:
Do you want to MAX on all rotations so you get to more accurate value over time ? (assuming this is coming from fixed value set by conmon)

Reply:
We don't care about max file size. We only care about bytes_logged and bytes_collected. I think this code can be simplified to accumulate directly into the prometheus client metrics (like the Go code at https://github.com/alanconway/file-metric/blob/master/cmd/file-metric/main.go) so that the embedded prom client has an up-to-date view whenever there is a pull from prometheus.

Reply:
Taken out this change as maxfsize need not be monitored as a metric.

Reply:
@eranra MAX over all rotation is not required as rotation always happen when fsize > maxsize of log file.